packages/core/src/wot-impl.ts

```diff
@@ -39,8 +40,20 @@ export default class WoTImpl {
         throw new Error("not implemented");
     }
 
+    /** @inheritDoc */
     async requestThingDescription(url: string): Promise<WoT.ThingDescription> {
-        throw new Error("not implemented");
+        const uriScheme = new URL(url).protocol.split(":")[0];
+        const client = this.srv.getClientFor(uriScheme);
+        const result = await client.requestThingDescription(url);
+
+        const value = ContentManager.contentToValue({ type: result.type, body: await result.toBuffer() }, {});
```

Comment:
Unrelated comment. I found it strange that the function ContentToValue does not accept directly the `result` variable which is type Content. Something to record in an issue for future improvements.

Reply:
Yeah, that is indeed a bit strange, apparently `contentToValue` expects input that implements the `ReadContent` interface which is not compatible with the regular `Content` class 🤔 It would definitely be better to have one unified `Content` API that is used across the codebase. Opening an issue for that sounds like a good idea to me!